tests/src/app-promotion.test.ts

```diff
@@ -0,0 +1,375 @@
+    // assert:  Bobs balance the same
+  });
+
+  it('will failed for non contract address', async () => {
+    // arrange: web3 creates new address - 0x0
+
+    // assert: Admin calls appPromotion.sponsorContract(0x0) throws
+    // assert: Admin calls appPromotion.sponsorContract(Substrate address) throws
+  });
+
+  it('will actually sponsor transactions', async () => {
+    // TODO test it because this is a new way of contract sponsoring
+  });
+});
+
+// STOP CONTRACT SPONSORING
+describe.skip('app-promotion stopSponsoringContract', () => {
+  it('will set contract sponsoring mode as disabled', async () => {
+    // arrange: Alice deploys Flipper
+    // arrange: Admin calls appPromotion.sponsorContract(Flipper.address)
+    
+    // act:     Admin calls appPromotion.stopSponsoringContract(Flipper.address)
+    // assert:  contract sponsoring mode = TODO
+
+    // act:     Bob calls Flipper
+
+    // assert:  PalleteAddress balance did not change
+    // assert:  Bobs balance less than before /// Bob payed some fee
+  });
+
+  it('can not be called by non-admin', async () => {
+    // arrange: Alice deploys Flipper
+    // arrange: Admin calls appPromotion.sponsorContract(Flipper.address)
+
+    // assert:  Random calls appPromotion.stopSponsoringContract(Flipper.address) throws
+    // assert:  contract sponsor is PallereAddress
+  });
+
+  it('will not affect a contract which is not sponsored by pallete', async () => {
+    // arrange: Alice deploys Flipper
+    // arrange: Alice sets self sponsoring for Flipper
+    
+    // act:     Admin calls appPromotion.stopSponsoringContract(Flipper.address) throws
+
+    // assert:  contract.sponsoringMode = Self
+    // assert:  contract.sponsor to be contract
+  });
+
+  it('will failed for non contract address', async () => {
+    // arrange: web3 creates new address - 0x0
+
+    // expect stopSponsoringContract(0x0) throws
+  });
+});
+
+describe.skip('app-promotion rewards', () => {
+  it('will start charging only after 24h', async () => {
+
+  });
+
+  it('will charge 0.05% for staking period', async () => {
+    // arrange: bob.stake(10000);
+    // arrange: bob.stake(20000);
+    // arrange: waitForRewards();
+
+    // assert:  bob.staked to equal [10005, 20010] //TODO 
+  });
+  
+  it('will be charged from treasury', async () => {
+    // TODO
+  });
+  
+  it('will not be charged for unstaked (reserved) balance', async () => {
+    // arrange: bob.stake(10000);
+    // arrange: bob.unstake(5000);
+    // arrange: waitForRewards();
+
+    // assert:  bob.staked to equal [5002.5] 
+  });
+
+  it('will not be charged for tokens reserved for non-staking reason', async() => {
+    // arrange: bob balance = 20000
+    // arrange: bob locks 10000 // TODO how?
+    // arrange: bob.stake(1000);
+    // arrange: waitForRewards();
+
+    // assert:  bob.staked to equal [1000.5] 
+  });
+
+  it('will bring compound interest', async () => {
+    // TODO will it?
```

Comment:
Yes, it will. The daily 0.05% is calculated based on the total amount staked, and credited to the staked portion, so there will be compound interest.